### DISCUSSION TITLE
hc:Pagination  获取的pageindex 值是上次的

### DISCUSSION THREAD

**Discussion by wsx-198611** [HandyOrg/HandyControl#1340] [General]
hc:Pagination  获取的pageindex值 是上次的，例如我有2页，当前页为第2页，点击第1页时触发事件，触发事件代码中CommandParameter获取到的PageIndex值却为历史值2
代码如下：
<hc:Pagination  Name="coolingIndx" HorizontalAlignment="Center" MaxPageCount="{Binding MaxCount}" PageIndex="1" IsJumpEnabled="True" DataCountPerPage="10" >
                                    <i:Interaction.Triggers>
                                        <i:EventTrigger EventName="PageUpdated">
                                            <i:InvokeCommandAction Command="{Binding PageUpdatedCommand}" CommandParameter="{Binding ElementName=coolingIndx, Path=PageIndex}"/>
                                        </i:EventTrigger>
                                    </i:Interaction.Triggers>
 </hc:Pagination>

**Comment by boogaloo-jw** [comment]
给你看下我是怎么用的哦
 `<hc:Pagination Name="PageControl" MaxPageCount="{Binding PageModel.MaxPageCount}" DataCountPerPage="{Binding PageModel.DataCountPerPage}" PageIndex="{Binding PageModel.PageIndex}" IsJumpEnabled="True">
                    <hc:Interaction.Triggers>
                        <hc:EventTrigger EventName="PageUpdated">
                            <hc:EventToCommand Command="{Binding PageChangeCommand}" PassEventArgsToCommand="True"/>
                        </hc:EventTrigger>
                    </hc:Interaction.Triggers>
                </hc:Pagination>

[RelayCommand]
        private void PageChange(FunctionEventArgs<int> info)
        {
            Users.Clear();
            var res = totalUserList.Skip((info.Info - 1) * this.PageModel.DataCountPerPage).Take(this.PageModel.DataCountPerPage);
            foreach (var item in res)
            {
                Users.Add(item);
            }
        }`
